Lib/test/test_typing.py

```diff
@@ -894,6 +906,93 @@ class C(Generic[Unpack[Ts]]): pass
         self.assertNotEqual(C[Unpack[Ts1]], C[Unpack[Ts2]])
 
 
+class TypeVarTuplePicklingTests(BaseTestCase):
+    # These are slightly awkward tests to run, because TypeVarTuples (and things
+    # made generic using TypeVarTuples) are only picklable if defined in the
+    # global scope. We therefore need to push various things defined in these
+    # tests into the global scope with `global` statements at the start of each
+    # test.
+
+    @all_pickle_protocols
+    def test_pickling_then_unpickling_results_in_equality(self, proto):
+        global Ts1  # See explanation at start of class.
+        Ts1 = TypeVarTuple('Ts1')
+        Ts2 = pickle.loads(pickle.dumps(Ts1, proto))
+        self.assertEqual(Ts1, Ts2)
```

Comment:
We should use `assertIs` instead to assert we get the exact same TVT back.

Reply:
Done. (I'm assuming `assertEqual` is still correct for the tuples, but re-open if I'm wrong.)